src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -929,6 +932,95 @@ internal string GetPropertyRawText()
             return _parent.GetPropertyRawValueAsString(_idx);
         }
 
+        public bool ValueEquals(string text)
+        {
+            return ValueEquals(text.AsSpan());
+        }
+
+        public bool ValueEquals(ReadOnlySpan<byte> utf8Text)
+        {
+            if (TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedStringComparison(TokenType);
+            }
+            return TextEqualsHelper(utf8Text);
+        }
+
+        public bool ValueEquals(ReadOnlySpan<char> text)
+        {
+            if (TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedStringComparison(TokenType);
+            }
+
+            Span<byte> otherUtf8Text;
+            byte[] otherUtf8TextArray = null;
+            ReadOnlySpan<byte> utf16Text = MemoryMarshal.AsBytes(text);
+
+            int length = checked(utf16Text.Length * JsonConstants.MaxExpansionFactorWhileTranscoding);
+            if (length > JsonConstants.StackallocThreshold)
+            {
+                otherUtf8TextArray = ArrayPool<byte>.Shared.Rent(length);
+                otherUtf8Text = otherUtf8TextArray;
+            }
+            else
+            {
+                // Cannot create a span directly since it gets passed to instance methods on a ref struct.
```

Comment:
I don't think this comment holds.  (It does for Utf8JsonReader, but not for JsonElement).